components/suggest/src/store.rs

```diff
@@ -205,12 +205,11 @@ impl SuggestStore {
 
     /// Dismiss a suggestion
     ///
-    /// Dismissed suggestions will not be returned again
-    ///
-    /// In the case of AMP suggestions this should be the raw URL.
+    /// To dismiss a suggestion, pass in its [Suggestion::dismissal_key].
+    /// Dismissed suggestions will not be returned again.
     #[handle_error(Error)]
-    pub fn dismiss_suggestion(&self, suggestion_url: String) -> SuggestApiResult<()> {
-        self.inner.dismiss_suggestion(suggestion_url)
+    pub fn dismiss_suggestion(&self, dismissal_key: String) -> SuggestApiResult<()> {
```

Comment:
Yup, I like the idea of taking a `Suggestion` enum as the argument so that `dismissal_key()` can be called without crossing the FFI boundary.